tools/gcs-bucket-mover/gcs_bucket_mover/bucket_mover_tester.py

```diff
@@ -0,0 +1,207 @@
+from gcs_bucket_mover import bucket_mover_service
+
+BUCKET_LOCATION = 'us'
+CUSTOM_ROLE_NAME = 'projects/my_source_prj/roles/TestRole'
+DEFAULT_KMS_KEY_NAME = 'projects/my_source_prj/locations/global/keyRings/my_ring/cryptoKeys/my_key'
+EMAIL_FOR_IAM = 'test@google.com'
+LOGGING_BUCKET = 'source_bucket-logs'
+LOGGING_PREFIX = 'prefix-'
+STORAGE_CLASS = 'STANDARD'
+TOPIC_NAME = 'my_topic'
+
+
+def set_up_test_bucket(conf):
+    """Sets up the test bucket, adds objects and assigns various settings.
+
+    It makes sure none of the buckets already exist, and then runs the main bucket mover service.
+
+    Args:
+        conf: the argparser parsing of command line options
+    """
+
+    #Load the environment config values set in config.sh and create the storage clients.
+    config = configuration.Configuration(conf)
+
+    with yaspin(text='TESTING: Cleanup source bucket') as spinner:
+        try:
+            _check_bucket_exists_and_delete(
+                spinner, config.source_storage_client, conf.bucket_name,
+                conf.source_project)
+        except exceptions.Forbidden:
+            try:
+                #Maybe the bucket already exists in the target project.
+                _check_bucket_exists_and_delete(
+                    spinner, config.target_storage_client, conf.bucket_name,
+                    conf.target_project)
+            except exceptions.Forbidden:
+                spinner.write('TESTING: Not allowed to access bucket {}'.format(
+                    conf.bucket_name))
+                spinner.fail('X')
+                raise SystemExit()
+
+        source_bucket = create_bucket(config.source_storage_client,
+                                      conf.bucket_name)
+        spinner.write(
+            '{} TESTING: Bucket {} created in source project {}'.format(
+                bucket_mover_service.CHECKMARK, conf.bucket_name,
+                conf.source_project))
+
+    _upload_blobs(source_bucket)
+
+    with yaspin(text='TESTING: Cleanup target bucket') as spinner:
+        _check_bucket_exists_and_delete(spinner, config.target_storage_client,
+                                        config.temp_bucket_name,
+                                        conf.target_project)
+
+
+def create_bucket(storage_client, bucket_name):
+    """Creates the test bucket.
+
+    Also sets up lots of different bucket settings to make sure they can be moved.
+
+    Args:
+        storage_client: The storage client object used to access GCS
+        bucket_name: The name of the bucket to create
+
+    Returns:
+        The bucket object that has been created in GCS
+    """
+
+    bucket = storage.Bucket(client=storage_client, name=bucket_name)
+    # Requester pays
+    bucket.requester_pays = False
+    # CORS
+    policies = bucket.cors
+    policies.append({'origin': ['/foo']})
+    policies[0]['maxAgeSeconds'] = 3600
+    bucket.cors = policies
+    # KMS Key
+    #bucket.default_kms_key_name = DEFAULT_KMS_KEY_NAME
```

Comment:
Remove commented code.

Reply:
Added a comment above to explain how it can be used